### PR TITLE
Remove weird seek call in `GameplayClockContainer`

### PR DESCRIPTION
**Pull request by peppy**
Hopefully not required anymore. This can actually cause incorrect seeks to occur, at random, during actual game and test execution.

Split out from #19828 for isolated testing (and more test runs).

### CODE REVIEW

File: osu.Game/Screens/Play/GameplayClockContainer.cs [hunk 1]
```diff
@@ -93,13 +93,7 @@ public virtual void Start()
             ensureSourceClockSet();
 
             if (!decoupledClock.IsRunning)
-            {
-                // Seeking the decoupled clock to its current time ensures that its source clock will be seeked to the same time
-                // This accounts for the clock source potentially taking time to enter a completely stopped state
-                Seek(FramedClock.CurrentTime);
```
Comment (smoogipoo) [inline osu.Game/Screens/Play/GameplayClockContainer.cs:99]: I also wonder if this was somehow related to audio devices taking time to pause? I'm not sure if that's a thing though... Because then it would be doing the seek for the only reason of seeking the track back to the current time.

Comment (peppy) [reply osu.Game/Screens/Play/GameplayClockContainer.cs:99]: Possibly. We'll have to see how this plays out in CI and local testing. I've been testing these changes only via tests so will do some actual gameplay tonight to test any non-tested edge cases.

**Review by smoogipoo — DISMISSED** [review]

**Comment by peppy** [comment]
At very least, we do seem to have test coverage of this (maybe indirectly)

JetBrains Rider 2022-08-18 at 10 38 13

Will investigate further.

**Comment by smoogipoo** [comment]
It looks to be related yeah.

Master:

https://user-images.githubusercontent.com/1329837/185375739-6f865c82-e1d4-4760-af0d-660024dd5431.mp4

This PR:

https://user-images.githubusercontent.com/1329837/185375760-159764bb-75fa-423d-b6f3-ca9a46a7eff2.mp4

Notice the jumping after unpause.

**Comment by peppy** [comment]
Closing and will keep as part of https://github.com/ppy/osu/pull/19828. The composite changes there do not have the failures seen here due to other related changes.
